### ISSUE TITLE
Source Value 8 is Obsolete

### ISSUE THREAD

**Issue by Doomsdayrs** [ATLauncher/ATLauncher#930]
### What happened?

As I was compiling ATLauncher for testing, I noticed that the `compileJava` task now reports that Java 8 compilation support will soon be removed.

This is with Gradle 8.7, Java SDK 22.

### Operating System

Windows

### Relevant log output

```shell
> Task :compileJava
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
```

### Is this actually a launcher bug?

- [X] I am sure this is a bug with the launcher and not an issue with running Minecraft

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by RyanTheAllmighty** [comment]
Yeah I noticed that in https://github.com/ATLauncher/ATLauncher/pull/911#issuecomment-2242912038

I think the legacy launch project needs to be converted to use something other than gradle as it really needs to support Java 7 source compilation (which currently it is set to 8 and causes issues with older instances using Java 7).

But Java 17 upgrade is going well, uptake has been better than I expected, so we might be able to move to Java 17 sooner than I thought, maybe within a month

![image](https://github.com/user-attachments/assets/36e391f3-4e22-4d1e-8d44-325b074a468d)

**Comment by RyanTheAllmighty** [comment]
I think ideally best way forward is going to be the following:

- Use the current upgrade to Java 17 screen but include a 10 second timer on the user ignoring it, so they are forced to view it and update the wording to tell them the launcher will no longer work soon, only once, if user ignores it, it won't show again
- After a couple weeks do the same as the above but on every launch of the launcher
- After a couple weeks don't allow the launcher to be started if not on Java 17 or newer. Same dialog as above, but if skipped it will close the launcher
- After a couple weeks release the Java 17 minimum update

Ideally wait a few weeks to see if the stats go up from above first, and if they do we can do this.

**Comment by Doomsdayrs** [comment]
> I think the legacy launch project needs to be converted to use something other than gradle as it really needs to support Java 7 source compilation (which currently it is set to 8 and causes issues with older instances using Java 7).

I do not think this is gradle, I think Java itself will be dropping support.

The warning does not appear when I use JDK 17, only when I use JDK 22

**Comment by Doomsdayrs** [comment]
Perhaps, it is alright to drop support for versions of Minecraft that depend on legacy launch?

**Comment by RyanTheAllmighty** [comment]
> I do not think this is gradle, I think Java itself will be dropping support.
> 
> The warning does not appear when I use JDK 17, only when I use JDK 22

Fantastic, that makes it a bit simpler


> Perhaps, it is alright to drop support for versions of Minecraft that depend on legacy launch?

Nah the legacy launch is a compiled jar that's [included in the launcher jar as a resource](https://github.com/ATLauncher/ATLauncher/blob/master/src/main/resources/legacy-launch-jar), it's compiled once and then never updated. Just needs to be recompiled to Java 7 and then updated in the jar and then forgotten about for many more years. It's simple enough to maintain, if the concern is having legacy code in this repo, happy to extract it out to a separate repository so this one is kept up to date
